### PR TITLE
Autodetect RocksDB storage.

### PR DESCRIPTION
**Pull request by AlekSi**

### CODE REVIEW

File: mysqld_exporter.go [hunk 1]
```diff
@@ -494,6 +494,19 @@ func (e *ExporterMr) scrape(ch chan<- prometheus.Metric) {
 	}
 	versionNum := getMySQLVersion(db)
 
+	var rocksDBEnabled bool
+	if *collectEngineRocksDBStatus == "auto" {
+		rocksDBEnabled, err = collector.RocksDBEnabled(db)
```
Comment (arvenil) [inline mysqld_exporter.go]: It makes sense to do this on every request, as in theory database or database capabilities can change while exporter is running... however this means `SHOW ENGINES` will be run every request, which might be not optimal. Just a comment to be aware of this, for now I think it's perfect solution.

File: collector/engine_rocksdb.go [hunk 2]
```diff
@@ -146,3 +146,29 @@ func ScrapeEngineRocksDBStatus(db *sql.DB, ch chan<- prometheus.Metric) error {
...
+	var dummy interface{}
+	for rows.Next() {
+		if err = rows.Scan(&engine, &support, &dummy, &dummy, &dummy, &dummy); err != nil {
+			return false, err
+		}
+
+		if engine != "ROCKSDB" {
+			continue
+		}
+		return support == "YES", nil
```
Comment (arvenil) [inline collector/engine_rocksdb.go]: Maybe also "DEFAULT"

File: mysqld_exporter.go (hunk 1 above)
Comment (AlekSi) [reply mysqld_exporter.go]: Yeah, I thought about it. There is another aspect: if the database changes significantly while exporter is running, then this and other scrapers will work incorrectly – we break [`prometheus.Collector` contract](https://github.com/prometheus/client_golang/blob/master/prometheus/collector.go#L16-L51) by changing what we send in `Collect`. But changing that is a significant amount of work.

File: collector/engine_rocksdb.go (hunk 2 above)
Comment (AlekSi) [reply collector/engine_rocksdb.go]: 👍

**Pull request merged**